subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -680,6 +680,12 @@ static void response_cb(struct http_response *rsp, enum http_final_call final_da
 				break;
 			}
 
+			if (rsp->http_status_code / 100 == 4) {
```

Comment:
A bit late in reviewing this, but I think using the `IN_RANGE(400, 499)` would be less cryptic